electricitymap/contrib/lib/models/events.py

```diff
@@ -93,16 +111,27 @@ def __setattr__(
         value: Optional[float],
     ) -> None:
         """
-        Overriding the setattr method to check for negative values and set them to None.
-        This method also keeps track of the modes that have been corrected.
+        Overriding the setattr method to check that the name is a valid production mode.
         """
         if not name in PRODUCTION_MODES:
             raise ValueError(f"Unknown production mode: {name}")
-        if value is not None and value < 0:
```

Comment:
You shouldn't remove this part.
Otherwise if someone calls a production mix like this:

```production_mix.wind =-1```
It will not be handled as we want it.